expression/builtin_time.go

```diff
@@ -2372,7 +2415,63 @@ type builtinSubTimeSig struct {
 // eval evals a builtinSubTimeSig.
 // See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_subtime
 func (b *builtinSubTimeSig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("SUB_TIME")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	if args[0].IsNull() || args[1].IsNull() {
+		return
+	}
+	var arg1 types.Duration
+	switch tp := args[1].Kind(); tp {
+	case types.KindMysqlDuration:
+		arg1 = args[1].GetMysqlDuration()
+	default:
+		s, err := args[1].ToString()
+		if err != nil {
+			return d, errors.Trace(err)
+		}
+		if getFsp(s) == 0 {
+			arg1, err = types.ParseDuration(s, 0)
+		} else {
+			arg1, err = types.ParseDuration(s, types.MaxFsp)
```

Comment:
@shenli 
if arg0 or arg1 is not  mysqltime or mysqlduration ,their fsp will be zero or the maxFsp.

mysql> select subtime(1,1.2);
+------------------+
| subtime(1,1.2)   |
+------------------+
| -00:00:00.200000 |
+------------------+
1 row in set (0.00 sec)


select subtime("1.2",1);
+------------------+
| subtime("1.2",1) |
+------------------+
| 00:00:00.200000  |
+------------------+
1 row in set (0.00 sec)